### PR TITLE
Add node:gulp subgenerator

### PR DESCRIPTION
**Pull request by SBoudrias**
TODO: tests are breaking because we don't have a good way to compare partial JSON. I'll fix this soon enough by adding a method to assert partial JS object match.

### CODE REVIEW

**Comment by hemanth** [comment]
:+1: Cool!

File: generators/gulp/templates/gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+gulp.task('static', function () {
+  return gulp.src('**/*.js')
+    .pipe(jshint('.jshintrc'))
+    .pipe(jshint.reporter('jshint-stylish'))
+    .pipe(jshint.reporter('fail'))
+    .pipe(jscs())
+    .on('error', handleErr);
+});
+
+gulp.task('test', function (cb) {
```
Comment (SBoudrias) [inline generators/gulp/templates/gulpfile.js]: Let's reorganize the code in there as `pre-test` and `test` tasks.

Comment (hemanth) [inline generators/gulp/templates/gulpfile.js]: `pre-test` will include?

Comment (SBoudrias) [inline generators/gulp/templates/gulpfile.js]: The coverage setup.

Comment (arthurvr) [inline generators/gulp/templates/gulpfile.js]: :+1:

File: test/gulp.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+'use strict';
+
```
Comment (arthurvr) [inline test/gulp.js]: remove that line :)

File: generators/gulp/templates/gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  var mochaErr;
+
+  gulp.src(['lib/**/*.js'])
+    .pipe(istanbul({includeUntested: true}))
```
Comment (arthurvr) [inline generators/gulp/templates/gulpfile.js]: > `{includeUntested: true}`

I'm confused by our style guide. Spaces or not?

File: test/gulp.js (hunk 2 above)
Comment (arthurvr) [inline test/gulp.js]: sorry for the nitpicking! Awesome work!

File: generators/gulp/templates/gulpfile.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    .pipe(jshint.reporter('jshint-stylish'))
+    .pipe(jshint.reporter('fail'))
+    .pipe(jscs())
+    .on('error', handleErr);
+});
+
+gulp.task('test', function (cb) {
+  var mochaErr;
+
+  gulp.src(['lib/**/*.js'])
```
Comment (arthurvr) [inline generators/gulp/templates/gulpfile.js]: > `['lib/**/*.js']`

no need to wrap in an array

File: generators/gulp/templates/gulpfile.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
+'use strict';
+var path = require('path');
+var gulp = require('gulp');
+var mocha = require('gulp-mocha');
+var jshint = require('gulp-jshint');
+var jscs = require('gulp-jscs');
+var istanbul = require('gulp-istanbul');<% if (includeCoveralls) { %>
+var coveralls = require('gulp-coveralls');<% } %>
+var plumber = require('gulp-plumber');
```
Comment (arthurvr) [inline generators/gulp/templates/gulpfile.js]: might just use [`gulp-load-plugins`](https://www.npmjs.com/package/gulp-load-plugins)?

Comment (SBoudrias) [inline generators/gulp/templates/gulpfile.js]: Nah, I hate these hacks. Let's keep it easy.

File: generators/gulp/templates/gulpfile.js (hunk 3 above)
Comment (kevva) [inline generators/gulp/templates/gulpfile.js]: I vote for no, but it seems like we're using different everywhere.

**Comment by SBoudrias** [comment]
Merged on master with fixes and passing tests.

**Comment by hemanth** [comment]
Super!

**Pull request closed without merging**